userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -86,33 +93,53 @@ engine::~engine()
 
 }
 
-int32_t engine::init(std::string socket_path)
+int32_t engine::init(std::string socket_path, std::string runsc_root_path, std::string trace_session_config_path)
 {
-	// Initialize the listen fd
-	int sock, ret;
-	struct sockaddr_un address;
-	unsigned long old_umask;
 	m_socket_path = socket_path;
-	if (m_socket_path.empty())
+	if(m_socket_path.empty())
 	{
 		strlcpy(m_lasterr, "Empty gVisor socket path", SCAP_LASTERR_SIZE);
 		return SCAP_FAILURE;
 	}
-
 	unlink(m_socket_path.c_str());
 
-	sock = socket(PF_UNIX, SOCK_SEQPACKET, 0);
+	if(runsc_root_path.empty())
```

Comment:
I think it'd be great to add a check to verify if the `runsc` binary exists and can be called (we can simply call `runsc --version`) at initialization time. This way we can avoid setting up sockets, threads and so forth if gvisor is not installed.